packages/kuma-gui/src/app/application/components/route-view/RouteView.vue

```diff
@@ -172,7 +173,21 @@ const routeView = {
     setAttrs([...attributes.values()])
   },
 }
-const routeParams = reactive<Params>(structuredClone(props.params) as Params)
+const routeParams = reactive<Params>({} as Params)
+
+onBeforeRouteUpdate((to, from) => {
+  if(route.name === props.name) {
+    // Check for certain query search params to be stored as global user preference
+    for (const key of ['size', 'format']) {
```

Comment:
I guess you might have thought this yourself, but ideally we should avoid referencing specific properties by name. I would guess we should be able to figure out which ones to reference based on whether they use `Number` or `String`.

I think the other standout thing is that this code seems to just check if the route param has changed, something that `watch` is made for. 🤔 I'm actually unsure about this one myself, because I'm wondering if I like using `onBeforeRouteUpdate` better even though we have to manually check for changes.

Leave this one with me, and I suppose we'll chat through this if we pair. I think at the very least tho (and I'd still be more than happy for this PR to go in without this improvement) we should avoid referring to the names of the properties

Reply:
Oh wait, this bit of code is always automatically saving things to localStorage whether I've called `route.update` or not?